spec/validations_spec.cr

```diff
@@ -21,6 +34,17 @@ private class TestValidationUser
     self
   end
 
+  def run_validations_with_message
+    validate_required city, state, message: "ugh"
+    validate_inclusions_of state, in: ["CA, NY"], message: "that one's not allowed"
+    validate_confirmation_of name, message: "name confirmation much match"
+  end
+
+  def run_validations_with_message_callables
+    validate_required city, state, message: ->(x : String, y : String){ "#{x} required message from Proc" }
```

Comment:
I like the idea of allowing a Proc or class, but I think making the names order dependent like this could be a bit confusing and/or lead to bugs. I'm thinking that if you want to use a proc for `validate_required` then you must do it with just one field at a time. 

```crystal
validate_required city, message: ->(city : String) { "#{city} required message from Proc"}
```

I think allowing a string message is fine with an array of fields though since that makes sense and wouldn't break if you change the field order.

The restriction would look like this I think:

```crystal
# Change the existing method signature
def validate_required(**fields, message : String = "is required")

# Add a new one for use with just one field 
def validate_required(field, message : Object)
```

Reply:
It might look more confusing since I called them x and y in the spec. The Proc arguments aren't city and state, they are the name and value relative to the field being validated. Let me know if I've interpreted your comment right. I can make the spec more clear on what it's doing by renaming the Proc args.